drivers/sdhc/sdhc_max32.c

```diff
@@ -0,0 +1,416 @@
+
+	sdhc_data->props.f_min = SDHC_CLOCK / (SDHC_SDHC_PCLK_DIV * SDHC_SDHC_MAX_DIV_VAL);
+	sdhc_data->props.f_max = SDHC_CLOCK;
+	sdhc_data->props.is_spi = 0;
+	sdhc_data->props.max_current_180 = 0;
+	sdhc_data->props.max_current_300 = 0;
+	sdhc_data->props.max_current_330 = 0;
+	sdhc_data->props.host_caps.timeout_clk_freq = 0x01;
+	sdhc_data->props.host_caps.timeout_clk_unit = 1;
+	sdhc_data->props.host_caps.sd_base_clk = 0x00;
+	sdhc_data->props.host_caps.max_blk_len = 0b10;
+	sdhc_data->props.host_caps.bus_8_bit_support = false;
+	sdhc_data->props.host_caps.bus_4_bit_support = false;
+	sdhc_data->props.host_caps.adma_2_support = true;
+	sdhc_data->props.host_caps.high_spd_support = true;
+	sdhc_data->props.host_caps.sdma_support = true;
+	sdhc_data->props.host_caps.suspend_res_support = true;
+	sdhc_data->props.host_caps.vol_330_support = true;
+	sdhc_data->props.host_caps.vol_300_support = true;
+	sdhc_data->props.host_caps.vol_180_support = true;
+	sdhc_data->props.host_caps.address_64_bit_support_v4 = false;
+	sdhc_data->props.host_caps.address_64_bit_support_v3 = false;
+	sdhc_data->props.host_caps.sdio_async_interrupt_support = true;
+	sdhc_data->props.host_caps.slot_type = 00;
+	sdhc_data->props.host_caps.sdr50_support = true;
+	sdhc_data->props.host_caps.sdr104_support = true;
+	sdhc_data->props.host_caps.ddr50_support = true;
+	sdhc_data->props.host_caps.uhs_2_support = false;
+	sdhc_data->props.host_caps.drv_type_a_support = true;
+	sdhc_data->props.host_caps.drv_type_c_support = true;
+	sdhc_data->props.host_caps.drv_type_d_support = true;
+	sdhc_data->props.host_caps.retune_timer_count = 0;
+	sdhc_data->props.host_caps.sdr50_needs_tuning = 0;
+	sdhc_data->props.host_caps.retuning_mode = 0;
+	sdhc_data->props.host_caps.clk_multiplier = 0;
+	sdhc_data->props.host_caps.adma3_support = false;
+	sdhc_data->props.host_caps.vdd2_180_support = false;
+	sdhc_data->props.host_caps.hs200_support = false;
+	sdhc_data->props.host_caps.hs400_support = false;
+	sdhc_data->props.power_delay = sdhc_config->power_delay_ms;
+}
+
+static int sdhc_max32_init(const struct device *dev)
+{
+	const struct sdhc_max32_config *sdhc_config = dev->config;
+	int ret = 0;
+	mxc_sdhc_cfg_t cfg;
+
+	ret = pinctrl_apply_state(sdhc_config->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0) {
+		LOG_ERR("Pinctrl apply error:%d", ret);
+		return ret;
+	}
+
+	ret = clock_control_on(sdhc_config->clock, (clock_control_subsys_t)&sdhc_config->perclk);
+	if (ret) {
+		LOG_ERR("Clock control on error:%d", ret);
+		return ret;
+	}
+
+	cfg.bus_voltage = sdhc_config->bus_volt;
+	cfg.block_gap = 0;
+	/* Maximum divide ratio, frequency must be 100 - 400 kHz during Card Identification"*/
+	cfg.clk_div = SDHC_SDHC_MAX_DIV_VAL;
+
+	ret = MXC_SDHC_Init(&cfg);
+	if (ret != E_NO_ERROR) {
+		LOG_ERR("MXC_SDHC_Init error:%d", ret);
+		return ret;
+	}
+
+	/* init delay, without it applications fail. 5ms found empirically */
+	k_sleep(K_MSEC(5));
```

Comment:
Is there some busy bit in the hardware we could poll here? Honestly, it sounds to me like what this delay is doing is giving the SD host controller time to send the 74 initialization clocks required at power on. That should not occur here though, it should occur each time the SD controller toggles power on to the card (in the `sdhc_max32_set_io` function)